import/index_java/src/org/vufind/index/FormatCalculator.java

```diff
@@ -605,12 +605,20 @@ protected List<String> getFormatsFrom33xFields(Record record) {
             String desc = getSubfieldOrDefault(typeField, 'a', "");
             String code = getSubfieldOrDefault(typeField, 'b', "");
             String source = getSubfieldOrDefault(typeField, '2', "");
+            if (desc.equals("computer program")) {
+                return List.of(); // rely on getFormatFromRecordType(), 008/26 is more precise
+            }
             if ((desc.equals("two-dimensional moving image") || code.equals("tdi")) && source.equals("rdacontent")) {
                 formats.add("Video");
                 if (isOnline) {
                     formats.add("VideoOnline");
                 }
             }
+            boolean computerOrCartographicDS = desc.equals("computer dataset") || desc.equals("cartographic dataset");
+            boolean crdOrCod = code.equals("crd") || code.equals("cod");
+            if (source.equals("rdacontent") && (computerOrCartographicDS || crdOrCod)) {
```

Comment:
I believe this should be:

```suggestion
            if (computerOrCartographicDS || (source.equals("rdacontent") && crdOrCod)) {
```

If the description matches, we don't care about the source... but if it's a code, we do need to take the source into account.

Reply:
It makes sense, but then shouldn't we do the same thing for "two-dimensional moving image" ?

Reply:
Good point, I misread the code on the second time through -- which is why I was asking about checking "rdacontent" consistently/globally the first time.

I think we need to make a decision: either just bail out of the function of it's not rdacontent on the assumption that we don't really know how it should be processed, or else restrict rdacontent checks to codes, on the assumption that plain-English descriptions mean the same thing regardless of the standard being used.

I don't have strong feelings on which is better -- one is more common-sense, the other is more cautious.

Reply:
OK. I prefer the common-sense one (restrict rdacontent checks to codes). It will give better results if the source is missing for some reason. I will prepare an update.

Reply:
Thanks, this looks reasonable to my eye now -- but I'll leave this open for a while to allow further discussion, and I'll try to find time to do a full index of Villanova's MARC records to check for consequences (either good or bad).